src/js/franklin-dashboard.js

```diff
@@ -39,119 +38,151 @@ angular
+}
+
+//bootstrap franklin-dashboard angular app
+function bootstrapApplication(response) {
+  angular.element(document).ready(function() {
+
+    franklinApp
+      .constant("GITHUB_CONFIG", response.data)
+      .constant('VERSION', packageJson.version)
+      .config(($authProvider, ENV, $stateProvider, $urlRouterProvider,
+        toastrConfig, $resourceProvider, $interpolateProvider) => {
+
+        $interpolateProvider.startSymbol('[[').endSymbol(']]');
+
+        //angular routing depending on login status
+        $stateProvider
+          .state('logged', {
+            url: '/dashboard',
+            templateUrl: 'dashboard/dashboard.html',
+            resolve: {
+              loginRequired: loginRequired
+            }
+          })
+          .state('logged.franklinRepos', {
+            url: '/',
+            templateUrl: 'dashboard/franklin-repos.html'
+          })
+          .state('logged.detailInfo', {
+            url: '/detail',
+            templateUrl: 'dashboard/repo-detail.html'
+          })
+          .state('logout', {
+            url: '/login',
+            templateUrl: 'login/login.html',
+            resolve: {
+              skipIfLoggedIn: skipIfLoggedIn
+            }
+          });
+
+        $urlRouterProvider.otherwise('/dashboard');
+
+        $resourceProvider.defaults.stripTrailingSlashes = false;
+
+        //Github login configuration
+        $authProvider.withCredentials = false;
+
+        //configure satellizer with client_id from franklin-api
+        $authProvider.github({
+          clientId: response.data.client_id,
+          url: ENV.FRANKLIN_API_URL + '/auth/github/',
+          //Ask permission for hooks, deploy keys, private repos
+          scope: ['user:email', 'admin:repo_hook', 'repo'],
+          redirectUri: window.location.origin
+        });
+
+
+        //toastr configuration - error messages
+        angular.extend(toastrConfig, {
+          positionClass: 'toast-bottom-center',
+          closeButton: false,
+          closeHtml: '<button>&times;</button>',
+          extendedTimeOut: 10000,
+          tapToDismiss: true,
+          timeOut: 10000,
+        });
+
+        //Auxiliar functions for routing
+        function skipIfLoggedIn($q, $auth) {
+          var deferred = $q.defer();
+          if ($auth.isAuthenticated()) {
+            deferred.reject();
+          } else {
+            deferred.resolve();
+          }
+          return deferred.promise;
         }
-      })
-      .state('logged.franklinRepos', {
-        url: '/',
-        templateUrl: 'dashboard/franklin-repos.html'
-      })
-      .state('logged.detailInfo', {
-        url: '/detail',
-        templateUrl: 'dashboard/repo-detail.html'
-      })
-      .state('logout', {
-        url: '/login',
-        templateUrl: 'login/login.html',
-        resolve: {
-          skipIfLoggedIn: skipIfLoggedIn
+
+        function loginRequired($q, $location, $auth) {
```

Comment:
I'd rethink this function a bit. It doesn't always resolve/reject a promise. Should probably refactor it to always resolve or reject. And let whomever calls the function decide whether or not to take you to the `/login` view.